torch/csrc/utils/python_dispatch.cpp

```diff
@@ -130,13 +156,15 @@ void initDispatchBindings(PyObject* module) {
     }, "", py::arg("dispatch") = "")
   ;
 
-  m.def("_dispatch_library", [](const char* kind, std::string name, const char* dispatch) {
+  m.def("_dispatch_library", [](const char* kind, std::string name, const char* dispatch, const char* file_and_line) {
+    HANDLE_TH_ERRORS
     return std::make_unique<torch::Library>(
       parseKind(kind),
       std::move(name),
       std::string(dispatch) == "" ? c10::nullopt : c10::make_optional(c10::parseDispatchKey(dispatch)),
-      "/dev/null",
+      file_and_line,
```

Comment:
this looks wrong, you have no assurance that the passed in `const char*` has static lifetime; if it gets deallocated you are sadface

Reply:
would passing the file name as a string and doing `reinterpret_cast<const char*>(std::move(file))` fix the issue?

Reply:
nope